metisp/pymetis/src/pymetis/classes/inputs/inputset.py

```diff
@@ -54,6 +57,17 @@ def __init__(self, frameset: cpl.ui.FrameSet, **kwargs):
         self.tag_parameters: dict[str, str] = {}        # A dict of all tunable parameters determined from tags
         self.frameset = frameset
 
+        make_snake = re.compile(r'(?<!^)(?=[A-Z])')
+        cut_input = re.compile(r'Input$')
+
+        # Now iterate over all refined Inputs, instantiate them and feed them the frameset to filter.
+        for (name, input_type) in inspect.getmembers(self.__class__,
```

Comment:
Usually, when I want to use `inspect`, I'm doing something wrong. So this line triggers me, and there is not enough information here to take away my fear that we are doing something wrong here too. Maybe it is fine though.

Perhaps you can iterate over `self.__class__.__dict__.items()` instead? But maybe that won't work for some reason? It would still be a bit hacky, but somewhat less, IMHO.

I think `inspect` is also pretty heavy, and maybe implementation dependent (not sure)?

Reply:
I am not sure. There are many possible approaches, this felt cleanest to me. Also we can go fully explicit.

What I both like and dislike the most is that it happens fully automatically. It can lead to disgusting bugs (like failing to match the name of the overridden class, which results in extra inputs (but that can be caught by looking at the man page)). Also code collocation is horrendous -- all the magic happens way too high in the hierarchy.